src/parcsr_ls/par_nongalerkin.c

```diff
@@ -724,20 +724,21 @@ hypre_NonGalerkinIJBufferCompress( HYPRE_MemoryLocation memory_location,
    HYPRE_Int                *indys     = hypre_CTAlloc(HYPRE_Int,  (*ijbuf_rowcounter),
                                                        HYPRE_MEMORY_HOST);
 
-   HYPRE_Int                i, j, duplicate, cnt_new, rowcounter_new, prev_row;
+   HYPRE_Int                i, duplicate, cnt_new, rowcounter_new;
    HYPRE_Int                row_loc;
-   HYPRE_BigInt             row_start, row_stop, row;
+   HYPRE_BigInt             row_start, row_stop, row, prev_row, j;
```

Comment:
row_start and row_stop should be HYPRE_Int  or  780, 781 and 785, 786 need to be cast to HYPRE_BigInt, since ijbuf_numcols is HYPRE_Int.

Reply:
Let me update, thanks!